### DISCUSSION TITLE
How to access Kiali rest api

### DISCUSSION THREAD

**Discussion by ktsakos** [kiali/kiali#3791] [Q&A]
There is a swagger.json description into the repo.

I have deployed kiali with istio. The Web UI is served on port 20001 on k8s cluster.

Testing some resource url existed in the swagger.json are not working.

In which port i have to access the API? Essentially i want to take access in the kiali backend.

**Comment by jmazzitelli** [comment]
FYI: while there is a swagger.json and you can utilize the backend API, just note that it is not designed or implemented for public consumption and there is no guarantee for backward compatibility or proper documentation.

That said, you access it on the same port as the console. I am pretty sure the context path is /api (i.e. `https://<host>:20001/api/<and the rest>`).

If the swagger doc isn't correct, you can let us know and we'll try to fix it - but again, realize this isn't for public consumption and it doesn't surprise me that there may be things that are wrong in that swagger doc.

**Reply by jshaughn** [reply]
Could you comment on your use case for the API?

**Reply by lucasponce** [reply]
+1 on previous comments.
Kiali API is considered private to the Kiali UI, so, it can change without notice between minor versions.

Having said that, if you open Kiali from a browser with the Debug tabs enabled, you'll see examples of calls between Kiali UI and Kiali backend.

But those are considered implementation details.

If we understand which is the use case, perhaps we may help here.

**Reply by hugeps** [reply]
> Could you comment on your use case for the API?

Simple:  Kiali contains rich health monitoring facility for Service Mesh. 
We would like to use same as a means of synthetic monitoring of the Service Mesh health.
E.g. by calling the API from synthetics monitoring probe.

**Reply by jmazzitelli** [reply]
As of today, the internal Kiali API is no longer being designed, implemented, nor documented for external consumption. There is no guarantee the API will not change version-to-version and there is no guarantee of backward compatibility. In short, the API is only for internal Kiali usage, not meant for external usage by third parties.
